src/jdk.net/share/classes/jdk/net/ExtendedSocketOptions.java

```diff
@@ -199,6 +199,28 @@ private ExtendedSocketOptions() { }
         = new ExtSocketOption<UnixDomainPrincipal>
             ("SO_PEERCRED", UnixDomainPrincipal.class);
 
+    /**
+     * Disable IP packet fragmentation.
+     *
+     * <p> The value of this socket option is a {@code Boolean} that represents
+     * whether the option is enabled or disabled. When {@code true} fragmentation
+     * of outgoing IPv4 and IPv6 packets does not occur. This option can only be used
+     * with datagram sockets. When set, care must be taken to limit outgoing packet
+     * sizes to the {@link java.net.NetworkInterface#getMTU() local MTU}. Depending
+     * on the implementation and the network interface, packets larger than the MTU
+     * may be sent or dropped silently or dropped with an exception thrown.
+     *
+     * @apiNote
+     * For IPv4 this option sets the DF (Do not Fragment) flag in the IP packet
+     * header. This instructs intermediate routers to not fragment the packet.
+     * IPv6 routers never fragment packets. Instead, fragmentation is handled
+     * by the sending and receiving nodes exclusively. Setting this option for
+     * an IPv6 socket ensures that packets to be sent are never fragmented, in
+     * which case, the local network MTU must be observed.
+     */
+    public static final SocketOption<Boolean> IP_DONTFRAGMENT =
```

Comment:
Just noticed that we are missing `@since 19`.

Reply:
Thanks for noticing. I logged [JDK-8319825](https://bugs.openjdk.org/browse/JDK-8319825)